swift/chacha/chachaTests/chachaTests.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  chachaTests.swift
+//  chachaTests
+//
+//  Created by Leon Johnson on 27/09/2020.
+//  Copyright © 2020 Leon Johnson. All rights reserved.
+//
+
+import CryptoKit
+import XCTest
+@testable import chacha
+
+class ChachaTests: XCTestCase {
+    
+    private var cha: ChaCha!
+    
+    override func setUp() {
+        cha = ChaCha()
+    }
+
+    override func tearDown() {
+        cha = nil
+    }
+    
+    
+    func testGenerateKey_RandomKey(){
+        let key:SymmetricKey = cha.generateKey()
+        XCTAssertNotNil(key)
+        XCTAssertEqual(key.bitCount, 256)
+    }
+    
+    
+    /// This unit test checks whether the data can be encrypted and then decrypted using the same key
+    func testEncryptionKeyIsSymmetric(){
+        // Given
+        let secret = "bottom evoke mask jar patch distance force invite senior soccer allow youth normal beauty joke live rebel charge merge episode abandon donor screen video"
+        print("The encrypted data is: \(secret)")
+        let encryptedSecret = secret.data(using: .utf8)
+        let key: SymmetricKey = cha.generateKey()
+        let keyAsData = key.withUnsafeBytes({
+            return Data(Array($0))
+        })
+        
+        // When
+        let sealedBox = try! cha.encrypt(secret: encryptedSecret!, key: keyAsData) //sealedbox as bytes
+        let openedBox = try! cha.decrypt(cipher_bytes: sealedBox!, key: keyAsData)
+        let decodedSecret = String(decoding: openedBox!, as: UTF8.self)
+        
+        // Then
+        XCTAssertEqual(secret, decodedSecret)
```

Comment:
This seems to check i/o values for the Swift code. Can we compare a static string copied from the JS unit tests to make sure Apple's implementation in CryptoKit is compatible to the JS library?

Reply:
@leonjohnson I meant to test if the ciphertext output `ct` from js following JS code can be decrypted using your swift implementation

```js
const ct = await chacha.encrypt(pt, key);
```

from

https://github.com/photon-sdk/photon-lib/blob/master/__tests__/unit/chacha.test.js#L27


Since the `IV` is prepended to the ciphertext in:

https://github.com/photon-sdk/photon-lib/blob/master/src/chacha.js#L38